### PR TITLE
🏃 e2e test uses official Kubernetes version

### LINKED ISSUE #612: e2e test should target Kubernetes official release
/kind bug

**What steps did you take and what happened:**

e2e test runs against Kubernetes CI version. Sometimes e2e test fails due to the unstable version.

**What did you expect to happen:**

e2e test runs against Kubernetes offitial version.

**Anything else you would like to add:**

CI version is installed by [preKubeadmCommands](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/hack/ci/e2e-conformance/e2e-conformance_patch.yaml.tpl#L18-L19)

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by hidekazuna**
**What this PR does / why we need it**:

We need e2e test to use official Kubernetes version instead of unstable CI version which is sometimes RC version and ends with error.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #612 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
e2e test use official Kubernetes version instead of CI version
```

### CODE REVIEW

**Comment by hidekazuna** [comment]
@jichenjc #605 , #607 should work after this PR.

File: hack/ci/e2e-conformance/e2e-conformance_patch.yaml.tpl [hunk 1]
```diff
@@ -15,86 +13,6 @@ spec:
         kubeletExtraArgs:
           v: "8"
     verbosity: 8
-    preKubeadmCommands:
```
Comment (jichenjc) [inline hack/ci/e2e-conformance/e2e-conformance_patch.yaml.tpl:18]: I didn't write this before, so not pay attention on this

why we remove all those code? covered by somewhere?
can you add those info into commit msg?

Comment (hidekazuna) [reply hack/ci/e2e-conformance/e2e-conformance_patch.yaml.tpl:18]: No,the purpose of this commit is removing those code actually. those code patch the manifest  and KUBERNETES_VERSION becomes CI version.
I updated the commit message.

**Pull request merged**